DRILL-5796: Added filter pushdown tests for files with multiple row g…

agozhiy opened the pull request:
…roups.

Also, reorganized filter pushdown tests, moved them to separate folders by categories.

Agirish commented on framework/resources/Functional/filter/pushdown/general/data.json:
```diff
@@ -0,0 +1,27 @@
+{
+  "testId": "filter_pushdown_data",
```
Can you please ensure the Test IDs across the PR are unique. I found this and the below file to share the IDs.

Agirish replied:
Thanks!

Agirish commented:
Can you please share details on what was reorganized and how is looks now. What are the new tests being added? What are being modified / moved? And some details on what the new tests are for?

agozhiy commented:
Before changes:
```
Functional/filter/pushdown
                  |_data
                  |_plan
                     |_regex
                     |_regex_no_order
```
After changes:
```
Functional/filter/pushdown
                  |_boolean_null
                  |_cast
                  |_complex_fields
                  |_general
                  |_item_star_operator
                  |_multiple_row_groups
                  |_union
```
Details:
* Added new tests to multiple_row_groups folder.
* Moved existing tests to separate folders depending on a feature tested.
* Combined data and plan verifications within single directories.
* Moved schema definitions from the queries to the json files (like dfs.drillTestDir).
* Changed extension of some tests to match the guidelines (*.sql.explain, *.plan)
